health-monitors/gpu-health-monitor/gpu_health_monitor/platform_connector/platform_connector.py

```diff
@@ -278,10 +278,20 @@ def send_health_event_with_retries(self, health_events: list[platformconnector_p
         metrics.health_events_insertion_to_uds_error.set(1.0)
         # Remove failed health events from entity cache
         for health_event in health_events:
-            for entity in health_event.entitiesImpacted:
-                cache_key = self._build_cache_key(health_event.checkName, entity.entityType, entity.entityValue)
+            if health_event.checkName == "GpuDcgmConnectivityFailure":
```

Comment:
Is this flow already tested in existing tests? If not, how can we add a test that fails without these changes, but passes with these changes?